src/Detector/DotNetCore/DotNetCorePlatformDetectorResult.cs

```diff
@@ -12,11 +14,38 @@ namespace Microsoft.Oryx.Detector.DotNetCore
     public class DotNetCorePlatformDetectorResult : PlatformDetectorResult
     {
         /// <summary>
-        /// Gets or sets the relative path to the location of the project file.
+        /// The relative path to the location of the project file.
         /// </summary>
         /// <example>
         /// "src/ShoppingWeb/ShoppingWeb.csproj"
         /// </example>
-        public string ProjectFile { get; set; }
+        public readonly string ProjectFile;
+
+        /// <summary>
+        /// Constructor of DotNetCorePlatformDetectorResult.
+        /// </summary>
+        /// <param name="platform"></param>
+        /// <param name="platformVersion"></param>
+        /// <param name="projectFile"></param>
+        public DotNetCorePlatformDetectorResult(string platform, string platformVersion, string projectFile) 
+            : base(platform, platformVersion)
+        {
+            this.ProjectFile = projectFile;
+        }
+
+        /// <summary>
+        /// Gets pairs of property names and values from the DotNetCorePlatformDetectorResult.
+        /// </summary>
+        /// <returns></returns>
+        public Dictionary<string, string> GetPlatformDetectorResultPropertyInfos()
```

Comment:
we should return `IReadOnlyDictionary` instead to make it immutable. Also, why can't we return an object here instead of a dictionary?

Reply:
+1 on what @gpcastro said. We do not need this method and should instead have plain properties.